src/main/java/core/basesyntax/model/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.model;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURE_SIZE = 100;
+    private static final int FIGURE_COUNT = 5;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(FIGURE_COUNT);
+        String randomColor = colorSupplier.getRandomColor();
+        double radius = randomSize();
+        double side = randomSize();
+        double height = randomSize();
+        double width = randomSize();
```

Comment:
let's call this method when passing values to constructors, and remove these variables